hubspot.js

```diff
@@ -6,3 +6,30 @@ script.defer = true
 script.src = '//js.hs-scripts.com/7230102.js'
 
 document.head.appendChild(script)
+
+function trackUrlChanges() {
+    const originalPushState = history.pushState
+    const originalReplaceState = history.replaceState
+
+    function handleUrlChange() {
+        setTimeout(function () {
+            // we need timeout to get  url and title (document.title) aligned for tracking, overwise the previous page title will be picked and associated to the latest url
+            var _hsq = (window._hsq = window._hsq || [])
+            const path = window.location.href.replace(window.location.origin, '')
+            _hsq.push(['setPath', path])
+            _hsq.push(['trackPageView'])
+        }, 100)
+    }
+
+    history.pushState = function (...args) {
+        originalPushState.apply(this, args)
+        handleUrlChange()
+    }
+
+    history.replaceState = function (...args) {
+        originalReplaceState.apply(this, args)
+        handleUrlChange()
+    }
+}
+
+trackUrlChanges()
```

Comment:
Does this fire on the first page view?

Reply:
yes, this function is executed on the page load, but it doesn't track the page view, but just setting tracking functionality.  However, this you question leaded me to double-check tracking on first load and yeah - there  was an issue described [here ](https://developers.hubspot.com/docs/reference/api/analytics-and-events/tracking-code#track-page-view) in documentation:  
> Please note:
> Calling this function manually before or during the initial page load could lead to duplicate views being tracked.

I've changed the code to avoid described above.